_posts/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment.md

```diff
@@ -0,0 +1,396 @@
+
+> info "Note"
+> It takes some time for the provisioning of the Controlplane, you can watch the process using some steps shared a bit later
+
+```console
+kubectl get KubeadmControlPlane ${CLUSTER_NAME} -n metal3
+kubectl describe KubeadmControlPlane ${CLUSTER_NAME} -n metal3
+```
+
+```yaml
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+kind: KubeadmControlPlane
+metadata:
+  [....]
+  ownerReferences:
+  - apiVersion: cluster.x-k8s.io/v1alpha3
+    blockOwnerDeletion: true
+    controller: true
+    kind: Cluster
+    name: bmetalcluster
+    uid: aec0f73b-a068-4992-840d-6330bf943d22
+  resourceVersion: "44555"
+  selfLink: /apis/controlplane.cluster.x-k8s.io/v1alpha3/namespaces/metal3/kubeadmcontrolplanes/bmetalcluster
+  uid: 99487c75-30f1-4765-b895-0b83b0e5402b
+spec:
+  infrastructureTemplate:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: Metal3MachineTemplate
+    name: bmetalcluster-controlplane
+    namespace: metal3
+  kubeadmConfigSpec:
+    files:
+    - content: |
+        [....]
+  replicas: 1
+  version: v1.18.0
+status:
+  replicas: 1
+  selector: cluster.x-k8s.io/cluster-name=bmetalcluster,cluster.x-k8s.io/control-plane=
+  unavailableReplicas: 1
+  updatedReplicas: 1
+```
+
+```console
+kubectl get Metal3MachineTemplate ${CLUSTER_NAME}-controlplane -n metal3
+```
+
+<br/>
+
+To track the progress of provisioning, you can try the following:
+
+```console 
+kubectl get BareMetalHosts -n metal3 -w
+```
+> The `BareMetalHosts` resource is created when `Metal³/metal3-dev-env` was deployed. It is a kubernetes resource that represents a Baremetal Machine, with all its details and configuration, and is managed by the `BareMetalOperator`. You can also use the short representation instead, i.e. `bmh` ( short for `BareMetalHosts`) in the command above.
+
+> You should see all the nodes that were created at the time of metal3 deployment, along with their current status as the provisioning progresses
+> info "Note"
+> All the baremetal hosts listed above were created when Metal³ was deployed in the *detailed metal3-dev-env walkthrough blogpost*.
+
+
+```console
+kubectl get Machine -n metal3 -w
+```
+> This shows the status of Machine associated with controlplane and we can watch the status of provisioning under PHASE 
+
+<br/>
+Once the provisioning is finished, let's get the host-ip : 
+
+```console
+sudo virsh net-dhcp-leases baremetal
+```
+> info "Note"
+> *baremetal is one of the 2 networks that were created at the time of Metal3 deployment, the other being “provisioning” which is used - as you have guessed - for provisioning the bare metal cluster. More details about networking setup in the metal3-dev-env environment are described in the - detailed metal3-dev-env walkthrough blogpost.*
+
+<br/>
+You can login to the master node if you want, and can check the deployment status
+
+```console
+ssh metal3@{master-node-ip}
```

Comment:
you can also use the VIP to log in (by default it is 192.168.111.249)

Reply:
Added it as an alternative